nestkernel/conn_builder.cpp

```diff
@@ -313,14 +331,16 @@ nest::ConnBuilder::connect()
 {
   // We test here, and not in the ConnBuilder constructor, so the derived
   // classes are fully constructed when the test is executed
-  if ( kernel().model_manager.connector_requires_symmetric( synapse_model_id_ )
-    and not( is_symmetric() or make_symmetric_ ) )
+  for ( auto syn_model : synapse_model_id_ )
```

Comment:
The following for loop is pretty long and complex. Can we break it up into helper methods?